### DISCUSSION TITLE
Optimistic Concurrency strategy raise exception or try to merge documents

### DISCUSSION THREAD

**Discussion by mdissel** [JasperFx/marten#2115] [Ideas]
Currently when a save conflict is detected an exception is raised (if optimistic concurrency is enabled). Maybe we could minimize a conflict situation by adding an option to the merge the conflicted documents into one document and only raise an exception when a conflict is found during merging. 
I think this option should have a documentstore default and can be overriden at document level or session level (or during SaveChanges(..) )

Something like this https://github.com/khalidsalomao/SimpleHelpers.Net/blob/master/docs/objectdiffpatch.md or https://github.com/wbish/jsondiffpatch.net could help with function

**Comment by jeremydmiller** [comment]
I'd vote to make that a completely semantically different operation. 

Sounds like a great pull request;)
